### PR TITLE
Assignment script for assigning sigs to individual E2Es

### LINKED ISSUE #42670: [proposal] restructuring tests to facilitate  SIG accountability
Per sig-testing today.

We generally agree:

- We'd like to restructure e2e to group by sig.
- We'd like to auto-associate every PR with a sig so there is global accountability as opposed to individual accountability for any regressions or flakes.
- We'd like to group every job/suite by sig.  possibly redundant to the first bullet.
- As part of this issue Jay has to curate the existing E2E tests with a rough first pass at sig responsibilities.   That should be fun :)

### PR DESCRIPTION
**Pull request by jayunit100**
**What this PR does / why we need it**:

to assign regressions to sigs xref =

**Which issue this PR fixes** *

fixes  #42670

### CODE REVIEW

**Comment by ixdy** [comment]
cc @kubernetes/sig-testing-pr-reviews

File: test/e2e/sigs.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+import glob
+import re
+from sets import Set
+
+go_files = glob.glob('*.go')
+
+sigs = {
+    "sig-api-machinery":["apiserver","admission","control","encoding","decoding"],
+    "sig-storage":["etcd","cassandra","gluster","ceph","nfs","volume"],
```
Comment (jeffvance) [inline test/e2e/sigs.py]: aplhabetic order would help when adding or deleting sigs

**Comment by ixdy** [comment]
/assign jeffvance

(Jeff handoff)

**Comment by ixdy** [comment]
/assign @jeffvance

(Jeff handoff)

File: test/e2e/sigs.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+            assignments[file_name]=("no-sig",0)
+
+        tags = Set([])
+        scores = {
+        }
+        for s in sigs:
+            scores[s]=0
+
+        for l in f:
+            for signal in ["It","By","should","the","per","create","Create","Describe","Sprint"]:
```
Comment (jeffvance) [inline test/e2e/sigs.py]: Context too? Ginkgo has synonyms for all of these too, right?

Comment (jayunit100) [inline test/e2e/sigs.py]: These are really heuristics that I played with to get a quick mapping that covered > 95% of the tests.  I think there could be other lines we would analyze.

**Comment by jeffvance** [comment]
@jayunit100 I see this informational _Jenkins verification_ log:
```
I0309 07:34:06.078] Boilerplate header is wrong for: /go/src/k8s.io/kubernetes/test/e2e/sigs.py
I0309 07:34:06.080] FAILED   hack/make-rules/../../hack/verify-boilerplate.sh	2s
I0309 07:34:06.080] Verifying hack/make-rules/../../hack/verify-cli-conventions.sh
```

**Comment by jayunit100** [comment]
fixed boilerplate

**Comment by jayunit100** [comment]
@jeffvance thanks yeah, the py boilerplate had a newline in it, should be ok now.

File: test/e2e/sigs.py (hunk 1 above)
Comment (jayunit100) [inline test/e2e/sigs.py]: agree on this.  i think there might be other suggestions on how this data structure should tag stuff, so , ill wait up to see what other tags people suggest.

File: test/e2e/assignments.txt [hunk 3]
```diff
@@ -0,0 +1,116 @@
+**************** Sig assignments for tests ********************
+monitoring.go ('sig-ui', 2)
+deployment.go ('sig-scheduling', 4)
```
Comment (timothysc) [inline test/e2e/assignments.txt]: There are a bunch of misfires here.  I think we need to let the sigs create the directory structure of have a e2e champion own the routing.

Comment (jayunit100) [inline test/e2e/assignments.txt]: I'm open to any algorithm for segregation, including manual curation - seems like that would be a pretty massive PR though ?

File: test/e2e/sigs.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    "sig-aws":["AWS", "Amazon", "S3"],
+    "sig-big-data":["Spark", "Kafka", "Hadoop", "Flink", "Storm", "Cassandra","Mongo","Elasticsearch"],
+    "sig-command-line-tools":["kubectl", "CLI"],
+    "sig-federation":["federation"],
+    "sig-cluster-lifecycle":["healthy","infrastructure","die","Disruptive"],
+    "sig-network":["DNS", "Ingress", "Proxy","Traffic"],
+    "sig-on-prem":["bare metal", "datacenter", "Baremetal"],
+    "sig-openstack":["openstack", "Swift", "Neutron"],
+    "sig-rkt":["rkt", "Rocket", "Coreos"],
+    "sig-scheduling":["scheduling", "performance", "scale", "affinity", "density", "critical","taint","toleration","tolerate"],
```
Comment (jayunit100) [inline test/e2e/sigs.py]: moving "performance" out of scheduling might be nice.  but not sure where that should go, or maybe just delete it entirely.

**Comment by jayunit100** [comment]
Closed per sig-testing meeting 3/14 : We are doing this by directory and  will be writing proposal up for that.

**Pull request closed without merging**